### PR TITLE
pki: make openssl database non-writable for others

### PR DESCRIPTION
**Pull request by michalskrivanek**
let's keep these files non-writable to any random user

Bug-Url: https://bugzilla.redhat.com/2088446

### CODE REVIEW

File: packaging/bin/pki-common.sh.in [hunk 1]
```diff
@@ -31,6 +31,7 @@ common_restore_perms() {
 	# of these files, so we have to reset
 	# our defaults
 	chown --reference="${pkidir}" "${pkidir}"/serial.txt* "${pkidir}"/database.txt* "${pkidir}"/.rnd* > /dev/null 2>&1
+	chmod go-wx "${pkidir}"/serial.txt* "${pkidir}"/database.txt* "${pkidir}"/.rnd* || die "Cannot set files permissions"
```
Comment (michalskrivanek) [inline packaging/bin/pki-common.sh.in:34]: so with umask this shouldn't actually be needed

**Comment by mz-pdm** [comment]
How about the other pki-* scripts -- shouldn't the umask be set there too?

**Comment by michalskrivanek** [comment]
> How about the other pki-* scripts -- shouldn't the umask be set there too?
AFAICT the ones that are called from engine-setup are all right, it's just ansible that calls it with 000

File: packaging/bin/pki-enroll-openssh-cert.sh [hunk 2]
```diff
@@ -32,6 +32,7 @@ sign() {
 		}
 		trap cleanup 0
 		cat "${PKIDIR}/private/ca.pem" > "${TMPCA}"
+		umask 0022
```
Comment (mz-pdm) [inline packaging/bin/pki-enroll-openssh-cert.sh:35]: Shouldn't it be before the openssl call above?

**Review by mz-pdm — COMMENTED** [review]
> AFAICT the ones that are called from engine-setup are all right, it's just ansible that calls it with 000

I see, OK.

File: packaging/bin/pki-enroll-openssh-cert.sh (hunk 2 above)
Comment (michalskrivanek) [reply packaging/bin/pki-enroll-openssh-cert.sh:35]: it's just the public key, but yes, it would be "nicer"

Comment (mz-pdm) [reply packaging/bin/pki-enroll-openssh-cert.sh]: There is no reason to have public key files writable by everybody.  And changing a public key can be used to assist with redirection to false hosts or accounts.

**Review by mz-pdm — APPROVED** [review]

**Comment by michalskrivanek** [comment]
this works on DISA STIG enabled system too, but the default umask there is 077 and so the created files are "more open" than before, e.g. for the CA (not affected by this patch) and the :
-rw-------. 1 root  root  1419 Jul 27 08:11 1000.pem
-rw-r--r--. 1 root  root  5442 Jul 27 08:11 1001.pem

**Comment by michalskrivanek** [comment]
@mz-pdm do you think it's worth fixing?
I can only think of following the default umask value by getting it from /etc/profile, e.g. using
$( . /etc/profile 2>/dev/null; command umask)

**Comment by mz-pdm** [comment]
You can get the current umask by simply running `umask` without arguments.  And yes, I think it's worth fixing -- while we should get the permissions right eventually, on the DISA STIG enabled systems it's good to be protected against occasional mistakes, e.g. when working with intermediate files in our code or in the tools we use.

Such a change would require verifying that we set permissions on all the resulting files explicitly (with `chmod`) and don't end up with too restrictive permissions on some of the files.  Maybe comparing permissions in OST with (while using the stricter umask in the system) and without this patch?

**Comment by mz-pdm** [comment]
> You can get the current umask by simply running umask without arguments.

Ah, yes, perhaps not in Ansible?  Even on those systems?  If so then I would say it's a bug in Ansible.

**Comment by mz-pdm** [comment]
> I can only think of following the default umask value by getting it from /etc/profile, e.g. using
> $( . /etc/profile 2>/dev/null; command umask)

I think a better way would be `sh -l -c umask`.  This should work even if Ansible sets its own umask.

File: packaging/bin/pki-enroll-openssh-cert.sh [hunk 3]
```diff
@@ -32,6 +32,7 @@ sign() {
 		}
 		trap cleanup 0
 		cat "${PKIDIR}/private/ca.pem" > "${TMPCA}"
+        umask "$(/bin/sh -l -c umask)"
```
Comment (mz-pdm) [inline packaging/bin/pki-enroll-openssh-cert.sh:35]: If there is no umask set explicitly in shell login scripts (/etc/profile, ...) then the umask is inherited from the calling environment. It shouldn't be a problem in RHEL-based systems though.

**Review by mz-pdm — APPROVED** [review]

**Review by mz-pdm — APPROVED** [review]

**Pull request merged**